pilot/pkg/model/context.go

```diff
@@ -852,34 +860,31 @@ func (node *Proxy) SetWorkloadLabels(env *Environment) {
 	}
 }
 
-// DiscoverIPVersions discovers the IP Versions supported by Proxy based on its IP addresses.
-func (node *Proxy) DiscoverIPVersions() {
-	for i := 0; i < len(node.IPAddresses); i++ {
-		addr := net.ParseIP(node.IPAddresses[i])
-		if addr == nil {
-			// Should not happen, invalid IP in proxy's IPAddresses slice should have been caught earlier,
-			// skip it to prevent a panic.
-			continue
-		}
-		if node.GlobalUnicastIP == "" && addr.IsGlobalUnicast() {
-			node.GlobalUnicastIP = addr.String()
-		}
-		if addr.To4() != nil {
-			node.ipv4Support = true
-		} else {
-			node.ipv6Support = true
-		}
+// DiscoverIPMode discovers the IP Versions supported by Proxy based on its IP addresses.
+func (node *Proxy) DiscoverIPMode() {
```

Comment:
I always has some confusion on this function, Istio discover IP Mode from `proxy.IPAddresses`, but how does it handle this situation? eg. a service is deployed in a dual-stack k8s kuberntes, but user define its ip family as below:
```
......
ipFamilies:
  - IPv6
  ipFamilyPolicy: SingleStack
......
```
This make thing become complicated, especially after k8s support dual-stack.

Reply:
It depends on what IP addresses proxy listens on and that is what we care about. BTW, it is not related to this PR, we will have to enhance it separately if we have to